plugins/woocommerce/includes/class-wc-remote-logger.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+use Automattic\WooCommerce\Utilities\FeaturesUtil;
+
+/**
+ * WooCommerce Remote Logger
+ *
+ * The WooCommerce remote logger class adds functionality to log WooCommerce errors remotely based on if the customer opted in and several other conditions.
+ *
+ * No personal information is logged, only error information and relevant context.
+ *
+ * @class WC_Remote_Logger
+ * @since 9.2.0
+ * @package WooCommerce\Classes
+ */
+class WC_Remote_Logger {
+	/**
+	 * Determines if remote logging is allowed based on the following conditions:
+	 *
+	 * 1. The feature flag for remote error logging is enabled.
+	 * 2. The user has opted into tracking/logging.
+	 * 3. The store is allowed to log based on the variant assignment percentage.
+	 * 4. The current WooCommerce version is the latest so we don't log errors that might have been fixed in a newer version.
+	 *
+	 * @return bool
+	 */
+	public function is_remote_logging_allowed() {
+		if ( ! FeaturesUtil::feature_is_enabled( 'remote_logging' ) ) {
+			return false;
+		}
+
+		if ( ! $this->is_user_opted_in() ) {
+			return false;
+		}
+
+		if ( ! $this->is_variant_assignment_allowed() ) {
+			return false;
+		}
+
+		if ( ! $this->is_latest_woocommerce_version() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if the user has opted into tracking/logging.
+	 *
+	 * @return bool
+	 */
+	private function is_user_opted_in() {
+		return 'yes' === get_option( 'woocommerce_allow_tracking', 'no' );
+	}
+
+	/**
+	 * Check if the store is allowed to log based on the variant assignment percentage.
+	 *
+	 * @return bool
+	 */
+	private function is_variant_assignment_allowed() {
+		$assignment = get_option( 'woocommerce_remote_variant_assignment', 0 );
+		return ( $assignment <= 12 ); // Considering 10% of the 0-120 range.
+	}
+
+	/**
+	 * Check if the current WooCommerce version is the latest.
+	 *
+	 * @return bool
+	 */
+	private function is_latest_woocommerce_version() {
+		$latest_wc_version = $this->fetch_latest_woocommerce_version();
+
+		if ( is_null( $latest_wc_version ) ) {
+			return false;
+		}
+
+		return version_compare( WC()->version, $latest_wc_version, '>=' );
+	}
+
+	/**
+	 * Fetch the latest WooCommerce version using the WordPress API and cache it.
+	 *
+	 * @return string|null
+	 */
+	private function fetch_latest_woocommerce_version() {
```

Comment:
> The current store WooCommerce version is the latest. This can ensure that we are not logging known issues that have been fixed in a later version.

I think it is useful to know how many customers are still experiencing known issues with outdated versions. 🤔  Any thoughts @woocommerce/ghidorah

Reply:
I just added Proton team as a reviewer to the PR to get their feedback on the changes. 🙂

Reply:
It sounds like a valid rollout strategy to limit the logging initially and gradually expand it.

Perhaps then, we don't need to make the REST API call but instead could just hardcode a cutoff timestamp for when errors should stop being logged. We would need to update this for every release until we have 100% rollout, however.

Reply:
Logging the latest version for now and expanding later sounds good to me 👍

Reply:
> Perhaps then, we don't need to make the REST API call but instead could just hardcode a cutoff timestamp for when errors should stop being logged. We would need to update this for every release until we have 100% rollout, however.

That could work, too! 👍  I prefer the REST API call because it's more dynamic, and we don't have to worry about updating it for every release. Besides, I think that we would still like to limit the logging ability to only some woocommerce versions, such as the latest minus two versions, to reduce unnecessary logging.  Since we cache the response, it shouldn't cause big performance issues. 🤔

Reply:
> I prefer the REST API call because it's more dynamic

That sounds good, @chihsuan.

Sidenote: could we use a similar technique to get the current WordPress version? For Core Profiler and other places in WC Admin, it would be good not to recommend plugins when the installed WP is too old. If so, it might be good to bring this out to a generic function for wider use.